src/abstractdatatable/abstractdatatable.jl

```diff
@@ -673,34 +673,34 @@ without(df::AbstractDataTable, c::Any) = without(df, index(df)[c])
 # Its first argument (currently) must be a DataTable.
 
 # catch-all to cover cases where indexing returns a DataTable and copy doesn't
-Base.hcat(df::AbstractDataTable, x) = hcat!(df[:, :], x)
+Base.hcat(dt::AbstractDataTable, x) = hcat!(dt[:, :], x)
 
-Base.hcat(df::AbstractDataTable, x, y...) = hcat!(hcat(df, x), y...)
+Base.hcat(dt::AbstractDataTable, x, y...) = hcat!(hcat(dt, x), y...)
 
 # vcat only accepts DataTables. Finds union of columns, maintaining order
-# of first df. Missing data become null values.
+# of first dt. Missing data become null values.
 
-Base.vcat(df::AbstractDataTable) = df
+Base.vcat(dt::AbstractDataTable) = dt
 
-Base.vcat(dfs::AbstractDataTable...) = vcat(AbstractDataTable[dfs...])
+Base.vcat(dts::AbstractDataTable...) = vcat(AbstractDataTable[dts...])
 
-Base.vcat(dfs::Vector{Void}) = dfs
-function Base.vcat{T<:AbstractDataTable}(dfs::Vector{T})
-    isempty(dfs) && return DataTable()
-    coltyps, colnams, similars = _colinfo(dfs)
+Base.vcat(dts::Vector{Void}) = dts
```

Comment:
https://github.com/JuliaData/DataTables.jl/pull/18

it passed tests for me on v0.5